### PR TITLE
[Auto Suggest] Add MDS Support Along with A Few Cleanup and tests

### PR DESCRIPTION
**Pull request by mengweieric**
### Description

- MDS support for OpenSearch SQL

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/data/public/antlr/opensearch_sql/code_completion.ts [hunk 1]
```diff
@@ -44,32 +43,33 @@ export const getSuggestions = async ({
...
   position,
   query,
-  connectionService,
-}: QuerySuggestionGetFnArgs): Promise<ISuggestionItem[]> => {
-  const { api } = getUiSettings();
+  openSearchDashboards,
+}: QuerySuggestionGetFnArgs): Promise<QuerySuggestion[]> => {
+  const { api } = openSearchDashboards.uiSettings;
+  const dataSetManager = openSearchDashboards.data.query.dataSet;
+  const selectedDataSet = dataSetManager.getDataSet();
```
Comment (ashwin-pc) [inline src/plugins/data/public/antlr/opensearch_sql/code_completion.ts:50]: can you use the `useDatasetManager` hook to get a stateful dataset. See here for reference: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data/public/ui/dataset_navigator/dataset_navigator.tsx#L83

File: src/plugins/data/public/antlr/shared/utils.test.ts [hunk 2]
```diff
@@ -6,54 +6,6 @@
 import { of } from 'rxjs';
 import { getRawSuggestionData$, fetchData, fetchTableSchemas, fetchColumnValues } from './utils';
 
-describe('getRawSuggestionData$', () => {
```
Comment (ashwin-pc) [inline src/plugins/data/public/antlr/shared/utils.test.ts:9]: Why are we deleting tests?

File: src/plugins/data/public/antlr/shared/utils.ts [hunk 3]
```diff
@@ -3,81 +3,44 @@
...
-      return from(dataSourceReuqstHandler({ dataSourceId, title }));
-    })
-  );
-
 export const fetchData = (
   tables: string[],
   queryFormatter: (table: string, dataSourceId?: string, title?: string) => any,
   api: any,
-  connectionService: any
+  dataSet: any
```
Comment (ashwin-pc) [inline src/plugins/data/public/antlr/shared/utils.ts:15]:
```suggestion
  dataSet: SimpleDataSet
```

File: src/plugins/data/public/antlr/shared/utils.ts [hunk 4]
```diff
@@ -3,81 +3,44 @@
...
+      reject(err);
+    }
   });
 };
 
 export const fetchTableSchemas = (
   tables: string[],
   api: any,
-  connectionService: any
+  selectedDataSet: any
```
Comment (ashwin-pc) [inline src/plugins/data/public/antlr/shared/utils.ts:43]:
```suggestion
  selectedDataSet: SimpleDataSet
```

**Review by ashwin-pc — COMMENTED** [review]
The rest looks good, just a small change to types and the way the dataset is fetched

File: src/plugins/data/public/antlr/opensearch_sql/code_completion.ts (hunk 1 above)
Comment (mengweieric) [reply src/plugins/data/public/antlr/opensearch_sql/code_completion.ts:50]: this is not a react component.

File: src/plugins/data/public/antlr/shared/utils.test.ts (hunk 2 above)
Comment (mengweieric) [reply src/plugins/data/public/antlr/shared/utils.test.ts:9]: this PR is not using observable, the tests are failing as these are testing against observable, changes are coming

**Comment by ashwin-pc** [comment]
removing the backport labels. The whole branch from main will be backported

**Review by ashwin-pc — APPROVED** [review]

**Review by kavilla — APPROVED** [review]

**Pull request merged**